benchmarks/base.py

```diff
@@ -18,6 +19,18 @@ def setup(self):
 
     def teardown(self):
         os.chdir(self.cwd)
+
+        if hasattr(self, "repo"):
+            self.repo.close()
+
+        if os.name == "nt":
+            # Windows does not allow to remove read only files
+            for root, _, files in os.walk(self.test_directory.name):
+                for file in files:
+                    path = f"{root}{os.sep}{file}"
```

Comment:
it is probably not worth optimizing here. Just use `join`, it is easier to read and maintain that way.